src/transformers/models/granite_speech/modeling_granite_speech.py

```diff
@@ -258,8 +257,8 @@ def __init__(self, config: GraniteSpeechEncoderConfig):
         # Precompute clamped relative positional encoding distances
         seq = torch.arange(config.context_size)
         relpos_dist = seq.view(-1, 1) - seq.view(1, -1)
-        self.attention_dists = torch.clamp(relpos_dist, -config.context_size, config.context_size) + config.max_pos_emb
-
+        attention_dists = torch.clamp(relpos_dist, -config.context_size, config.context_size) + config.max_pos_emb
+        self.register_buffer("attention_dists", attention_dists, persistent=False)
```

Comment:
registering the buffer automatically passes this to cuda, no need to re-move from cpu to gpu every forward pass

Reply:
that is a good find!